processor/metricstarttimeprocessor/README.md

```diff
@@ -8,3 +8,45 @@
 
 [development]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/component-stability.md#development
 <!-- end autogenerated section -->
+
+## Description
+
+The metric start time processor (`metricstarttime`) is used to set the start
+time for metric points with a cumulative aggregation temporality. It is
+commonly used with the `prometheus` receiver, which usually produces metric
+points without a start time.
+
+## Configuration
+
+Configuration allows configuring the strategy used to set the start time.
+
+```yaml
+processors:
+    # processor name: metricstarttime
+    metricstarttime:
+
+        # specify the strategy to use for setting the start time
+        strategy: true_reset_point
```

Comment:
What are other strategies that you envision this processor supporting? using another metric's value for the start time (`process_start_time_seconds`)? Wouldn't those also need to use the `true_reset_point` strategy to deal with resets?

Reply:
If a metric's value resets, it seems like we should always reset the start time, but I don't think that should involve inserting a "true reset" point.  I think it should be implicit for all adjustment strategies (maybe with an on/off config option, similar to fallback to collector start time?).

Reply:
I looked at https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#resets-and-gaps hoping to find a term for the other common mode -- but didn't find one.

If I had to make one up, it would be "missing_start_point" or similar, which is nearly identical logic to the presentation in this PR, except the value is unchanged. The first time the processor sees the series w/ a missing start time, it will enter the first timestamp it sees as the start time w/ unchanged value. Subsequent resets will be detected using the heuristic for cumulative values.

Reply:
I think this option should only determine how to handle the initial missing start point.  If we know a series has reset (e.g. because the value decreases), we should transition to a generic "use the most recent point's start timestamp going forward" handler.